test/commands/code-analyzer/init.test.ts

```diff
@@ -0,0 +1,78 @@
+import {stubSfCommandUx} from '@salesforce/sf-plugins-core';
+import {TestContext} from '@salesforce/core/lib/testSetup';
+import InitCommand from '../../../src/commands/code-analyzer/init';
+import {InitAction} from '../../../src/lib/actions/InitAction';
+
+describe('`code-analyzer init` tests', () => {
+	const $$ = new TestContext();
+	let spy: jest.SpyInstance;
+	let receivedActionInput: object;
+
+	beforeEach(() => {
+		stubSfCommandUx($$.SANDBOX);
+		spy = jest.spyOn(InitAction.prototype, 'execute').mockImplementation((input) => {
```

Comment:
Do you know if this prevents the execution of the action as well as spying on its inputs?

I typically don't use spy frameworks like this because it doesn't actually prevent the execution from going further. And if you aren't careful, you may get in trouble with the global state (of how it messes with prototypes behind the scenes). Basically, i'm curious what happens when we actually make the action do something... will it cause the test to run all of that unnecessarily.

Instead, I typically just create an interface (like an Action interface) that you can inject and have a SpyImplementation of that interface. Or I'll create a subclass of the thing i'm spying on and override its method alltogether to just spy instead of executre... again requires dependency injection (in which there are a number of ways of doing this, easiest is to add a constructor to all of your SfCommands that by default construct the runtime actions... but for testing can have the spy actions passed in).

Reply:
Yeah, I experimentally verified that the `mockImplementation()` part replaces the method's normal implementation with the one you provide, which in my case is a no-op.